e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts

```diff
@@ -0,0 +1,211 @@
+
+    await personalFiles.dataTable.selectItem(file3);
+    await personalFiles.acaHeader.shareButton.click();
+
+    const urlBefore = await personalFiles.shareDialog.getLinkUrl();
+    await personalFiles.shareDialog.shareToggle.click();
+    await personalFiles.confirmDialog.cancelButton.click();
+    expect(await personalFiles.shareDialog.isShareToggleChecked()).toBe(true);
+
+    const urlAfter = await personalFiles.shareDialog.getLinkUrl();
+    expect(urlBefore).toEqual(urlAfter);
+  });
+
+  test('[C306999] Unshare a file from the context menu', async ({ personalFiles, searchPage, nodesApiAction, page }) => {
+    await personalFiles.acaHeader.searchButton.click();
+    await searchPage.searchInput.searchButton.click();
+    await searchPage.searchOverlay.checkFilesAndFolders();
+    await searchPage.searchOverlay.searchFor(file4);
+
+    await personalFiles.dataTable.rightClickOnItem(file4);
+    await personalFiles.matMenu.clickMenuItem('Shared Link Settings');
+    const url = await personalFiles.shareDialog.getLinkUrl();
+    await personalFiles.shareDialog.shareToggle.click();
+
+    await personalFiles.confirmDialog.okButton.click();
+    expect(await nodesApiAction.isFileShared(file4Id)).toBe(false);
+
+    await page.goto(url);
+    await personalFiles.viewer.waitForViewerToOpen();
+    expect(personalFiles.viewer.fileTitleButtonLocator).not.toEqual(file4);
+
+    await page.reload();
+  });
+
+  test('[C306998] Consumer - on Search Results - file shared by other user', async ({ personalFiles, searchPage, nodesApiAction }) => {
+    await personalFiles.acaHeader.searchButton.click();
+    await searchPage.searchInput.searchButton.click();
+    await searchPage.searchOverlay.checkFilesAndFolders();
+    await searchPage.searchOverlay.searchFor(fileSite1);
+
+    await personalFiles.dataTable.selectItem(fileSite1);
+    await personalFiles.acaHeader.shareButton.click();
+
+    expect(await personalFiles.shareDialog.isShareToggleChecked()).toBe(true);
+    await personalFiles.shareDialog.shareToggle.click();
+    await personalFiles.confirmDialog.okButton.click();
+    expect(await nodesApiAction.isFileShared(fileSite2Id)).toBe(true);
+  });
+
+  test('[C307000] Consumer - on Search Results - file shared by the user', async ({ personalFiles, searchPage, nodesApiAction }) => {
+    await personalFiles.acaHeader.searchButton.click();
+    await searchPage.searchInput.searchButton.click();
+    await searchPage.searchOverlay.checkFilesAndFolders();
+    await searchPage.searchOverlay.searchFor(fileSite2);
+
+    await personalFiles.dataTable.selectItem(fileSite2);
+    await personalFiles.acaHeader.shareButton.click();
+    expect(await personalFiles.shareDialog.isShareToggleChecked()).toBe(true);
+
+    await personalFiles.shareDialog.shareToggle.click();
+    await personalFiles.confirmDialog.okButton.click();
+    expect(await nodesApiAction.isFileShared(fileSite2Id)).toBe(false);
```

Comment:
we can add step to very on UI like right click on file menu should be 'Shared Link Settings',
or add some timeout before check on backend to avoid flaky test.

Reply:
Added tiny timeout before check on backend, thanks!